nab/detectors/naive/naive_detector.py

```diff
@@ -0,0 +1,46 @@
+# ----------------------------------------------------------------------
+# Copyright (C) 2015, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""
+This is implementation of the "naive forecast" is a baseline algorithm for time-series
+forecasting. It repeats the last seen value. So `P(t+1) = P(t)`.
+"""
+
+from nab.detectors.base import AnomalyDetector
+
+class NaiveDetector(AnomalyDetector):
+
+
+  def initialize(self):
+    super().initialize()
+    self.predicted = 0.0 #previous value, last seen
+
+  def handleRecord(self, inputData):
+    """The anomaly score is simply the last seen value"""
+    current = float(inputData["value"])
+    inputData['predicted'] = self.predicted
+    anomalyScore = 1 #FIXME how compute anomaly score from predicted, current?
```

Comment:
Looking for help: this is the implementation of naive predictor. 
How do I compute anomaly score from the 2 numbers?

Reply:
You could use something like 1-exp(-nx) where n is a hyperparameter (for ex n = 10) and x = abs(current - predicted)/predicted. Other options could be arctan, 1-n/(x+1), etc